### PR TITLE
feat: sarif action

### PR DESCRIPTION
**Pull request by gal-legit**
#### What's being changed?
- legitify action: add code scanning support (using sarif).
- legitify action: upload json and sarif outputs to workflow artifacts.
- legitify action: hide output if the repository is private (only upload code scanning results).
- legitify action: add compile_legitify option to build legitify from source.
- minor fixes to markdown and sarif.

#### Is this PR related to an existing issue?
X

#### Check off the following:

- [x] This PR follows the CONTRIBUTION.md guidelines
- [x] I have self-reviewed my changes before submitting the PR

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -22,6 +22,63 @@ inputs:
...
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # ratchet:actions/setup-go@v3
+      with:
+        go-version: 1.19
+    - id: compile_legitify
+      if: ${{ inputs.compile_legitify == 'true' }}
+      shell: bash
+      env:
+        ref: "${{ inputs.legitify_base_version }}"
+      run: |
+        git clone https://github.com/Legit-Labs/legitify.git legitify-build
```
Comment (noamd-legit) [inline action.yml:51]: instead of full clone, download only the needed ref.

File: action.yml [hunk 2]
```diff
@@ -22,6 +22,63 @@ inputs:
...
+    - id: analyze
+      shell: bash
+      env:
+        github_token: ${{ inputs.github_token }}
+        analyze_self_only: ${{ inputs.analyze_self_only }}
+        repositories: ${{ inputs.repositories }}
+        legitify_base_version: ${{ inputs.legitify_base_version }}
+        scorecard: ${{ inputs.scorecard }}
+        upload_code_scanning: ${{ inputs.upload_code_scanning }}
+        compile_legitify: ${{ inputs.compile_legitify }}
```
Comment (noamd-legit) [inline action.yml:64]:
```suggestion
        pre_installed: ${{ inputs.compile_legitify }}
```

Comment (noamd-legit) [reply action.yml:64]: Maybe extract the download part from the node action to the composite instead of passing a flag

File: index.js [hunk 3]
```diff
@@ -25,26 +25,76 @@ async function uploadErrorLog() {
...
+
+    // generate a markdown version for the action output
+    myError = ""
+    options.listeners.stdout = (data) => {
+      myOutput += data.toString();
+    }
+    const convertMarkdownArgs = ["convert", "--input-file", jsonFile, "--output-format", "markdown"]
+    console.log("execute legitify convert markdown:", convertMarkdownArgs)
+    await exec.exec('"./legitify"', convertMarkdownArgs, options)
+    if (isPrivate) {
```
Comment (noamd-legit) [inline index.js:90]: We need a way to demonstrate legitify capabilities on a public repository

File: internal/outputer/formatter/formatter_sarif.go [hunk 4]
```diff
@@ -21,6 +21,21 @@ func newSarifFormatter() OutputFormatter {
 	}
 }
 
+func (f *sarifFormatter) URIFromLink(link string) (base, uri string) {
```
Comment (nadav-legit) [inline internal/outputer/formatter/formatter_sarif.go:24]: use url.Parse(link) to get the parts

**Review by nadav-legit — APPROVED** [review]

**Pull request merged**